runtime/executor/src/main/java/org/apache/nemo/runtime/executor/data/DirectByteBufferOutputStream.java

```diff
@@ -106,24 +125,30 @@ public void write(final byte[] b) {
    * @param   len   the number of bytes to write.
    */
   @Override
-  public void write(final byte[] b, final int off, final int len) {
+  public void write(final byte[] b, final int off, final int len) throws IOException {
     int byteToWrite = len;
     int offset = off;
-    while (byteToWrite > 0) {
-      if (currentBuf.remaining() <= 0) {
-        newLastBuffer();
-        currentBuf = dataList.getLast();
-      }
-      final int bufRemaining = currentBuf.remaining();
-      if (bufRemaining < byteToWrite) {
-        currentBuf.put(b, offset, bufRemaining);
-        offset += bufRemaining;
-        byteToWrite -= bufRemaining;
-      } else {
-        currentBuf.put(b, offset, byteToWrite);
-        offset += byteToWrite;
-        byteToWrite = 0;
+    try {
+      while (byteToWrite > 0) {
+        if (currentBuf.remaining() <= 0) {
+          newLastBuffer();
+          currentBuf = dataList.getLast();
+        }
+        final int bufRemaining = currentBuf.remaining();
+        if (bufRemaining < byteToWrite) {
+          currentBuf.put(b, offset, bufRemaining);
+          offset += bufRemaining;
+          byteToWrite -= bufRemaining;
+        } else {
+          currentBuf.put(b, offset, byteToWrite);
+          offset += byteToWrite;
+          byteToWrite = 0;
+        }
       }
+    } catch (IllegalAccessException e) {
+      throw new IOException("Not allowed for non-sequential MemoryChunk");
+    } catch (MemoryAllocationException e) {
```

Comment:
I believe this is never invoked, since we never actually instantiate a 'MemoryAllocationException', at least in the current PR.
(Same goes for other similar if/else statements)